2010-09-17  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Fstab.pm: Don't resolve LVM devices via readlink; the
		underlying canoncial path names aren't stable across reboots (thanks Brian
		Kroth for suggesting a similar patch).
Index: trunk/lib/setup-storage/Fstab.pm
===================================================================
--- trunk.orig/lib/setup-storage/Fstab.pm
+++ trunk/lib/setup-storage/Fstab.pm	
@@ -248,21 +248,7 @@
         # skip entries without a mountpoint
         next if ($l_ref->{mountpoint} eq "-");
 
-        # real device name
-        my @fstab_key = ();
-
-        # resolve the symlink to the real device
-        # and write it as the first entry
-        &FAI::execute_ro_command("readlink -f /dev/$device/$l", \@fstab_key, 0);
-
-        # remove the newline
-        chomp ($fstab_key[0]);
-
-        # make sure we got back a real device
-        ($FAI::no_dry_run == 0 || -b $fstab_key[0]) 
-          or die "Failed to resolve /dev/$device/$l\n";
-
-        my $device_name = $fstab_key[0];
+        my $device_name = "/dev/$device/$l";
 
         # if the mount point the /boot mount point, variables must be set
         if ($l_ref->{mountpoint} eq $boot_mnt_point) {
